### PR TITLE
init ui and handle event trending music

### PR DESCRIPTION
**Pull request by thaohq-0814**
Init ui and handle event trending music
Link: https://edu-redmine.framgia.vn/issues/5471
image

### CODE REVIEW

File: app/src/main/res/layout/fragment_home.xml [hunk 1]
```diff
@@ -1,5 +1,70 @@
...
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@+id/textView"
+                />
+            <TextView
+                android:id="@+id/textView2"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="16dp"
+                android:layout_marginTop="24dp"
```
Comment (buivanhieu7112) [inline app/src/main/res/layout/fragment_home.xml]: hardcode

**Review by buivanhieu7112 — APPROVED** [review]

**Comment by thaohq-0814** [comment]
ready

File: app/src/main/java/com/framgia/music_50/screen/home/HomeFragment.java [hunk 2]
```diff
@@ -31,7 +39,31 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         return view;
     }
 
+    @Override
+    public void onGetTrendingTracksSuccess(List<Track> list) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_50/screen/home/HomeFragment.java]:
```suggestion
    public void onGetTrendingTracksSuccess(List<Track> tracks) {
```

File: app/src/main/java/com/framgia/music_50/screen/home/HomeContract.java [hunk 3]
```diff
@@ -1,12 +1,19 @@
 package com.framgia.music_50.screen.home;
 
+import com.framgia.music_50.data.model.Track;
 import com.framgia.music_50.screen.BasePresenter;
+import java.util.List;
 
 public interface HomeContract {
     interface View {
+        void onGetTrendingTracksSuccess(List<Track> list);
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_50/screen/home/HomeContract.java]:
```suggestion
        void onGetTrendingTracksSuccess(List<Track> tracks);
```

File: app/src/main/java/com/framgia/music_50/screen/home/adapter/TrendingTrackAdapter.java [hunk 4]
```diff
@@ -1,35 +1,88 @@
...
 
     @Override
     public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
+        viewHolder.bindViewData(mTrackList.get(i));
     }
 
     @Override
     public int getItemCount() {
-        return 0;
+        return mTrackList.size();
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_50/screen/home/adapter/TrendingTrackAdapter.java]: check null size
```suggestion
        return  mTrackList !=null ? mTrackList.size() : 0;
```

File: app/src/main/java/com/framgia/music_50/screen/home/adapter/TrendingTrackAdapter.java [hunk 5]
```diff
@@ -1,35 +1,88 @@
...
+            mListener = listener;
+            mArtistNameTextView = itemView.findViewById(R.id.textViewArtistName);
+            mArtworkImageView = itemView.findViewById(R.id.imageViewArtwork);
+            mTrackTitleTextView = itemView.findViewById(R.id.textViewTrackTitle);
+            itemView.setOnClickListener(this);
         }
 
         @Override
         public void onClick(View v) {
+            mListener.onItemClickListener(mTrackList.get(getAdapterPosition()));
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/music_50/screen/home/adapter/TrendingTrackAdapter.java]: check null mListener

File: app/src/main/res/layout/fragment_home.xml [hunk 6]
```diff
@@ -1,5 +1,70 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        >
+        <android.support.constraint.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            >
+            <TextView
+                android:id="@+id/textView"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_home.xml]: đặt tên có nghĩa e

File: app/src/main/res/layout/fragment_home.xml [hunk 7]
```diff
@@ -1,5 +1,70 @@
...
+                android:layout_height="wrap_content"
+                android:layout_marginTop="@dimen/trending_music_recycler_margin_top"
+                android:orientation="horizontal"
+                app:layoutManager="android.support.v7.widget.LinearLayoutManager"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@+id/textView"
+                />
+            <TextView
+                android:id="@+id/textView2"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/fragment_home.xml]: as above

**Comment by thaohq-0814** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**